src/lib/metrics/Metrics.ts

```diff
@@ -280,12 +282,12 @@ export class Metrics {
 
   handleWindowCounter = (event: Event) => {
     const { detail } = event as CustomEvent;
-    this.pushCounter(detail.event, detail.data);
+    this.pushCounter<any>(detail.event, detail.data);
```

Comment:
This part is pretty confusing, as in `pushCounter/pushTiming` we're expecting pretty strict structure, but in the only one call of `pushCounter/pushTiming` we're manually pass `<any>`, not quite understood why even update pushCounter/pushTiming signature typings

Reply:
I've updated typing to require specific event name and params, plan to store all types of events in one place in lib/metrics/. It's pretty hard to walk the code and look for all of possible events

in this place **any** is used because this function is window event handler that uses to handle custom window events emitted by emitMetricEvent / emitMetricCounter / emitMetricTiming -  which is typed in the same way as pushEvent / pushCounter / pushTiming